src/commands/oas2tson.js

```diff
@@ -0,0 +1,149 @@
+import { Command } from 'commander'
+import filenamify from 'filenamify'
+import fs from 'fs-extra'
+import _get from 'lodash.get'
+import _trimStart from 'lodash.trimstart'
+import path from 'path'
+import pino from 'pino'
+import { exit } from 'process'
+import YAML from 'yaml'
+import $RefParser from '@apidevtools/json-schema-ref-parser'
+
+import { fromSchema } from '../utils/openapi-schema-to-json-schema-wrapper.cjs'
+import prettier from 'prettier'
+
+const COMPONENT_REF_REGEXP = /#\/components\/schemas\/[^"]+/g
+let outputSchemasMetaData = []
+
+export const adaptSchema = (generatedSchema, name, filename) => {
+  delete generatedSchema.$schema
+  generatedSchema.title = name
+  generatedSchema.$id = `${filename}.json`
+
+  if (generatedSchema.format?.includes('date')) {
+    generatedSchema.tsType = 'Date'
+  }
+}
+
+const processSchema = (schema, schemasPath, definitionKeyword, isArray) => {
+  Object.entries(schema).forEach(([key, value]) => {
+    // for elements in an array the name would be its index if we were
+    // to just use its key, so go into the parsed schema and get the
+    // actual name so the files are more easily identifiable
+    const name = isArray ? value.name : key
+    const filename = _trimStart(filenamify(name, { replacement: '-' }), '-')
+
+    adaptSchema(value, name, filename)
+
+    let schemaAsString = JSON.stringify(value, null, 2)
+    // N.B. - this obviously only supports refs where the string contains 'components/schemas'
+    // if we want to support refs in places other than this, we'll need to revisit this
+    // approach to be more flexible
+    const refs = schemaAsString.match(COMPONENT_REF_REGEXP)
+    refs?.forEach(ref => {
+      let refName = ref.split('/').slice(-1)
```

Comment:
refName is never reused, why is a let?

Are we sure that there's a linter configured for this project?